Fix Environment Extraction Pipeline

quazi-h opened the pull request:
## Why
The environment pipeline is not extracting any records due to some bad filter logic.
[Relevant ticket](https://broadinstitute.atlassian.net/browse/dspdc-1874)

## This PR
Refactored the logic to generate arms for the environment pipeline.
There is no "annual_" prefix for environment events and also months June, July, and September have a slightly different date format.

## Checklist
- [ ] Documentation has been updated as needed.

quazi-h committed:
Refactored the logic to generate arms for the environment pipeline.
There is no "annual_" prefix for environment events and also months June, July, and September have a slightly different date format.

quazi-h committed:
Refactor date formatting for alternate months (June, July, September).

quazi-broad commented on dap-etl/extraction/src/main/scala/org/broadinstitute/monster/dap/common/OkWrapper.scala:
```diff
@@ -26,7 +26,7 @@ class OkWrapper extends HttpWrapper {
           logger.info(s)
         }
-      }).setLevel(Level.BASIC))
+      }).setLevel(Level.BODY))
```
Was this change just for debugging? Would this affect anything? Perhaps storage size of logs?

quazi-broad replied:
Updated to HEADERS

quazi-broad commented on dap-etl/extraction/src/main/scala/org/broadinstitute/monster/dap/environment/EnvironmentExtractionPipeline.scala:
```diff
@@ -47,7 +49,16 @@ object EnvironmentExtractionPipeline extends ScioApp[Args] {
+    dateList
+      .map(date =>
+        // event names for June, July, and Sept have 4 alpha characters before the year
+        if (List(6, 7, 9).contains(date.getMonthValue)) {
```
What I wrote feels very hacky - please feel free to suggest and revisions of this logic

aherbst-broad replied:
Given the non-standard abbreviation used by the surveys, I think the alternative here is to use a `DateTimeFormatterBuilder`,  see [this](https://stackoverflow.com/questions/30816016/how-to-parse-non-standard-month-names-with-datetimeformatter) stackoverflow answer.

Something like this:
```
  val monthMap: java.util.Map[java.lang.Long, String] = Map(
      Long.box(4L) -> "Apr",
      Long.box(6L) -> "June",
      Long.box(7L) -> "July",
      Long.box(9L) -> "Sept"
// ... we'll need all 12 months here
    ).asJava
    val formatter = new DateTimeFormatterBuilder()
      .appendText(ChronoField.MONTH_OF_YEAR, monthMap)
      .appendPattern("yyyy")
      .toFormatter()

    val april = OffsetDateTime.parse("2021-04-03T10:15:30+01:00")
    val june = OffsetDateTime.parse("2021-06-03T10:15:30+01:00")
    val july = OffsetDateTime.parse("2021-07-03T10:15:30+01:00")
    val sept = OffsetDateTime.parse("2021-09-03T10:15:30+01:00")
```

This may be marginally cleaner at the expense of our own custom months map. Worth a try.

quazi-broad replied:
Implemented a DateTimeFormatterBuilder based on the provided example.
The new logic passed the existing unit tests:
`testOnly org.broadinstitute.monster.dap.environment.EnvironmentExtractionPipelineSpec`

quazi-h committed:
Updated unit test cases.
Caught a bug in eols_death logic - corrected and added a unit test case for it.

quazi-h committed:
Merge branch 'master' into qh-dspdc-1874-env-refresh-changes

quazi-h committed:
Reverting a logging change that was made for testing.
Added a custom DateTimeFormatterBuilder instead of mapping the nonstandard months for a cleaner implementation.

aherbst-broad commented on dap-etl/extraction/src/main/scala/org/broadinstitute/monster/dap/environment/EnvironmentExtractionPipeline.scala:
```diff
@@ -13,7 +15,9 @@ class EnvironmentExtractionFailException() extends Exception
 object EnvironmentExtractionPipeline extends ScioApp[Args] {
 
-  val formatter = DateTimeFormatter.ofPattern("MMMyyyy")
+  //val formatter = DateTimeFormatter.ofPattern("MMMyyyy")
```
Remove these commented out lines.

quazi-broad replied:
whoops, thanks

aherbst-broad reviewed (approved):
Some commented out code to remove, otherwise lgtm.

quazi-h committed:
Removing old code
